src/integration/java/io/pinecone/integration/dataplane/PineconeClientLiveIntegTest.java

```diff
@@ -24,14 +25,21 @@
 public class PineconeClientLiveIntegTest {
 
     private static final Logger logger = LoggerFactory.getLogger(PineconeClientLiveIntegTest.class);
+
+    private static PineconeConnection connection;
     private static VectorServiceGrpc.VectorServiceBlockingStub blockingStub;
 
     @BeforeAll
     public static void defineConfig() throws IOException, InterruptedException {
-        PineconeConnection connection = createIndexIfNotExistsDataPlane(3, IndexModelSpec.SERIALIZED_NAME_POD);
+        connection = createIndexIfNotExistsDataPlane(3, IndexModelSpec.SERIALIZED_NAME_POD);
         blockingStub = connection.getBlockingStub();
     }
 
+    @AfterAll
```

Comment:
I added `@AfterAll` in a bunch of these tests to make sure we're closing the `PineconeConnection` once we're done.

Reply:
I like this!!